src/facdb.py

```diff
@@ -90,58 +91,108 @@ def count_comparison(df, width=1000, height=1000):
             width=width,
             height=height,
             xaxis=dict(title="Percentage"),
-            xaxis_tickformat="%",
+            xaxis_tickformat=".2%",
             template="plotly_white",
             margin=go.layout.Margin(l=0, r=0, b=0, t=0),
         )
         st.plotly_chart(fig, config=dict({"scrollZoom": True}))
 
+    def generate_hover(measure):
+        if measure[0] == "mapped_pct_diff":
+            return measure[2] + " %{customdata[0]} <extra></extra>"
+        elif measure[0] == "pct_mapped_old":
+            return measure[2] + "%{customdata[0]}/%{customdata[2]} <extra></extra>"
+        elif measure[0] == "pct_mapped_new":
+            return measure[2] + "%{customdata[0]}/%{customdata[1]} <extra></extra>"
+
+    def plot_diff_table(df, rows_to_style, as_pct=False):
+        colors = px.colors.qualitative.Plotly
+        to_display = df.copy()
+
+        styled = (
+            to_display.style.applymap(
+                lambda x: f"background-color: {colors[0]}"
+                if x
+                else "background-color: white",
+                subset=rows_to_style[0:1],
+            )
+            .applymap(
+                lambda x: f"background-color: {colors[1]}"
+                if x
+                else "background-color: white",
+                subset=rows_to_style[1:2],
+            )
+            .applymap(
+                lambda x: f"background-color: {colors[2]}"
+                if x
+                else "background-color: white",
+                subset=rows_to_style[2:3],
+            )
+        )
+
+        if as_pct:
+            styled = styled.format({r: "{:.2%}".format for r in rows_to_style})
+
+        st.dataframe(styled)
+
     def by_classification():
         """
         qc_diff visualization
         """
-        thresh = st.sidebar.slider(
-            "difference threshold", min_value=0, max_value=300, value=5, step=1
-        )
         level = st.sidebar.selectbox(
             "select a classification level",
             ["datasource", "factype", "facsubgrp", "facgroup", "facdomain"],
             index=0,
         )
+        st.sidebar.success(
+            """
+            Use the dropdown input bar to select an attribute to review
+            """
+        )
+        dff = qc_diff.groupby(level).sum()
+        thresh = st.sidebar.number_input(
+            "difference threshold",
+            min_value=0,
+            max_value=dff["diff"].max() - 1,
+            value=5,
+            step=1,
+        )
 
         st.sidebar.success(
             """
-            Use the slide bar and drop down to change the difference
-            threshold and select the attribute to review
+            Use the number input bar to change the difference
+            threshold
             """
         )
-        st.header(f"Change in number of records by {level}")
+        st.header(f"Change in total number of records by {level}")
         st.write(f"diff > {thresh}")
```

Comment:
I think this description might be better. diff > 100 look weird to me when -100 also shows up. Absolute(diff) might be too incomprehensible also. How about "absolute change greater than 100"

Reply:
good comment, I will make this change now